examples/lock-app/nrfconnect/README.md

```diff
@@ -1,53 +1,101 @@
+    -   [Using Native shell](#using-native-shell)
+    -   [Supported nRF Connect SDK versions](#supported-nrf-connect-sdk-versions)
+-   [Configuring the example](#configuring-the-example)
+-   [Flashing and debugging](#flashing-and-debugging)
+-   [Accessing the command line](#accessing-the-command-line)
 
 <hr>
 
-<a name="intro"></a>
+<a name="overview"></a>
 
-## Introduction
+## Overview
 
-![nrf52840 DK](../../platform/nrfconnect/doc/images/nrf52840-dk.jpg)
+This example application is running on the nRF Connect platform, which is based
+on the
+[nRF Connect SDK](https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/index.html)
+and [Zephyr RTOS](https://zephyrproject.org/). Visit CHIP's
+[nRF Connect Platform Overview](TODO:...) to read more information about
+platform structure and dependencies.
 
-The nRF52840 lock example application provides a working demonstration of a
-connected door lock device, built using CHIP, and the Nordic nRF Connect. The
-example supports remote access and control of a simulated door lock over a
-low-power, 802.15.4 Thread network. It is capable of being paired into an
-existing CHIP network along with other CHIP-enabled devices. The example targets
-the
-[Nordic nRF52840 DK](https://www.nordicsemi.com/Software-and-Tools/Development-Kits/nRF52840-DK)
-development kit, but is readily adaptable to other nRF52840-based hardware.
+CHIP device running lock application is controlled by the CHIP controller device
+over Thread protocol. It is assumed that CHIP device by default has Thread
+disabled, so it should be paired with CHIP controller and get configuration from
+it. There are few actions necessary to make before full communication is
+possible and they were described below. There is also a test mode, which allows
+to start Thread with default settings, by pressing button manually, but it does
+not fully guarantee that device will be able to communicate with CHIP controller
+and other devices.
+
+### BLE Advertising
+
+After powering up device for the first time, it should start advertising over
+BLE in order to inform other devices about its presence. For security reasons
+different devices may start advertising automatically or on User demand and in
+this example case, starting is triggered by pressing button.
+
+### BLE Rendezvous
+
+In CHIP there is a commissioning procedure called Rendezvous, which is done over
+BLE between CHIP device and CHIP controller, performing the role of the
+commissioner. To start it, controller has to previously get onboarding
+information from the CHIP device. Data are encoded as a QR code payload and
+typically presented on a device's display or like in this example case, shared
+using NFC tag.
+
+### Thread Provisioning
+
+Successfully finishing Rendezvous procedure allows to perform Provisioning
+operation, which goal is to send Thread network credentials from CHIP controller
+to the CHIP device. As a result, device is able to attach the Thread network and
```

Comment:
"attach to" or "join"?

Reply:
Right, usually we used "join Thread network".